Automated Tests for Actions/Reducers/Components

Chiu-Kevin-bah opened the pull request:
- Added automated unit tests for `./src/actions`
- Added automated unit tests for `./src/components`
- Added automated unit tests for `./src/reducers`

![image](https://user-images.githubusercontent.com/1786875/43377100-c4255fb6-938c-11e8-919c-63804eaf3284.png)

Chiu-Kevin-bah committed:
- Renamed `actions/firstAction.js` => `actions/index.js`
- Renamed `reducers/firstReducer.js` => `reducers/index.js`

Chiu-Kevin-bah committed:
Redux Architecture

- Scaled redux architecture for easier/quicker bootstrapping of reducers and actions

Chiu-Kevin-bah committed:
Automated Units

- Added automated unit tests for `./src/actions`
- Added automated unit tests for `./src/components`
- Added automated unit tests for `./src/reducers`

Chiu-Kevin-bah committed:
Merge branch 'master' into feature/redux-api-tests

Chiu-Kevin-bah committed:
Update package.json

- Remove containers from coverage

mjoyce91 commented on package.json:
```diff
@@ -38,10 +38,25 @@
     "eslint-plugin-react": "^7.10.0",
     "sinon": "^6.1.3"
   },
+  "devDependencies": {
```
Combine with other `devDependencies` object in package.json

mjoyce91 replied:
Also I don't think `yarn install` is picking these up.

mjoyce91 commented on src/Components/Details.js:
```diff
@@ -21,4 +21,4 @@ const Details = ({ data, id }) => (
 );
 
-export default Details;
+export default DetailsComponent;
```
`Details` to match declaration

Chiu-Kevin-bah replied:
Good call! I fixed the `HomePage` one too

mjoyce91 commented on src/tests.js:
```diff
@@ -0,0 +1,48 @@
+// object. Calling each property should trigger the call count on the spy,
+// which we test. We reset call count between tests since we're using the
+// calledOnce method.
+export function testDispatchFunctions(mapDispatchToProps, config = {}) {
```
This looks familiar :grinning:

Chiu-Kevin-bah replied:
Why reinvent the wheel? 😉

mjoyce91 reviewed (dismissed):
See comments

mjoyce91 commented on src/Components/index.test.js:
```diff
@@ -0,0 +1,21 @@
+describe('Components', () => {
+  let i;
+
+  for(i = 0; i < files.length; i++) {
```
`files.forEach`?

Chiu-Kevin-bah replied:
Yea I always had `forEach` async issues....I dunno why...but I don't with `map`....is that weird?

At least when it's on the server-side. I don't actually remember async issues on client side

Chiu-Kevin-bah replied:
Maybe it's related to this?
https://coderwall.com/p/kvzbpa/don-t-use-array-foreach-use-for-instead

I started with a `forEach` and the test finished complaining about having no tests.

Being slower? I'm sure if this is true, u can't tell most of the time. Most of the time when I experience this it isn't a small array either so shrug.

Chiu-Kevin-bah committed:
- Fix unit tests and devDependencies merge issue

Chiu-Kevin-bah committed:
Coverage Enhancement

- Enhance coverage to 100%
- Added automated redux container test

Chiu-Kevin-bah committed:
- Update naming consistency `HomePageContainer` => `HomePageComponent`

Chiu-Kevin-bah committed:
- Due to nature of entry validation for api config, this should never be hit, so removing for the sake of simplicity

Chiu-Kevin-bah commented on src/api.js:
```diff
@@ -1,7 +1,7 @@
 export const config = {
-  baseURL: localStorage.getItem('API_URL') || 'http://jsonplaceholder.typicode.com',
+  baseURL: localStorage.getItem('API_URL'),
 };
```
Just wanted to note this change: Due to what ended up doing in the entry file for verifying `api.config`, `API_URL` wouldn't ever be undefined unless `api.config` had a bad value => In which case well I dunno lol. 

I think this eliminates the tracing baggage ^^ lol

mjoyce91 committed:
Linting update, test coverage, accessibility
